src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -2237,6 +2238,22 @@ FixedwingPositionControl::tecs_update_pitch_throttle(const hrt_abstime &now, flo
 	tecs_status_publish();
 }
 
+void FixedwingPositionControl::publishLocalPositionSetpoint(const position_setpoint_s current_waypoint)
+{
+	if (_global_local_proj_ref.isInitialized()) {
+		Vector2f current_setpoint = _global_local_proj_ref.project(current_waypoint.lat, current_waypoint.lon);
+		vehicle_local_position_setpoint_s local_position_setpoint{};
+		local_position_setpoint.x = current_setpoint(0);
```

Comment:
populate the unused fields with NANs?

`local_position_setpoint.vx = NAN`, etc

Reply:
Alternatively does populating any other existing data make sense (for casual interest)?
 - acceleration from L1?
 - velocities from TECS (remove wind?)

Reply:
I think for now IMO there is not much more to add - e.g. velocity setpoints are not velocity but airspeed setpoints / accelerations are not actual acceleration setpoints but roll setpoints. I think we can add these by cleaning up the controller one by one

Reply:
Some stuff - e.g. lateral acceleration- will be published with the new L1 controller (https://github.com/PX4/PX4-Autopilot/pull/18810) in npfg_status. I would also find it nicer if we publish whatever we can in local_position_setpoint. That though also requires some controller interface alignment - using local coordinates prob is one thing of that.